hydra/test_utils/config_source_common_tests.py

```diff
@@ -60,6 +69,21 @@ def test_is_config(
         ret = src.is_config(config_path=config_path)
         assert ret == expected
 
+    @mark.parametrize(  # type: ignore
+        "config_path, expected",
+        [
+            ("dataset", True),
+        ],
+    )
+    def test_is_config_with_overlap_name(
+        self, type_: Type[ConfigSource], path: str, config_path: str, expected: bool
+    ) -> None:
+        if self.skip_overlap_config_path_name():
+            pytest.skip("Plugin does not support overlap config name with config path.")
```

Comment:
add the plugin name (or type) to the error.
maybe something like:
```python
f"ConfigSourcePlugin {type_.__name__} does not support overlap config name with config path"
```

Also, this reads  better to me:
ConfigSourcePlugin X does not support config objects and config groups with overlapping names.